src/main/java/org/folio/rest/impl/ModTenantAPI.java

```diff
@@ -37,12 +33,16 @@ public class ModTenantAPI extends TenantAPI {
   @Autowired
   private StorageCleanupService storageCleanupService;
 
-  public ModTenantAPI() { //NOSONAR
+  @Autowired
+  private SystemUserAuthService systemUserAuthService;
+
+  public ModTenantAPI() {
     SpringContextUtil.autowireDependencies(this, Vertx.currentContext());
   }
 
   @Override
   Future<Integer> loadData(TenantAttributes attributes, String tenantId, Map<String, String> headers, Context context) {
+    systemUserAuthService.initializeSystemUser(headers);
```

Comment:
it seems that systemUserAuthService.initializeSystemUser(headers) should be invoked using the vertx.executeBlocking() method in order not to block event loop thread since systemUserAuthService service uses a blocking http client. 
Why it was decided to use this http client instead of http client provided by vertx?